src/feedspora/generic_client.py

```diff
@@ -182,75 +151,77 @@ def shorten_url(self, the_url):
                 if isinstance(exception, AssertionError):
                     all_shorteners = ' '.join(shortener.available_shorteners)
                     logging.error('URL shortener %s is unimplemented!',
-                                  self._url_shortener)
+                                  self._account['url_shortener'])
                     logging.info('Available URL shorteners: %s',
                                  all_shorteners)
                 else:
-                    logging.error('Cannot shorten URL %s with %s: %s', the_url,
-                                  self._url_shortener, str(exception))
+                    logging.error('Cannot shorten URL %s with %s: %s',
+                                  the_url, self._account['url_shortener'],
+                                  str(exception))
                 to_return = the_url
             # pylint: enable=broad-except
 
         return to_return
 
-    # pylint: disable=too-many-branches
+    def set_option_defaults(self, account):
+        '''
+        Set default values for options common to all clients
+        :param account:
+        '''
+
+        if 'max_posts' not in account:
+            self._account['max_posts'] = 0
+        if 'max_tags' not in account:
+            self._account['max_tags'] = 100
+        if 'post_prefix' not in account:
+            self._account['post_prefix'] = ''
+        if 'post_suffix' not in account:
+            self._account['post_suffix'] = ''
+        if 'post_include_content' not in account:
+            self._account['post_include_content'] = False
+        if 'post_include_media' not in account:
+            self._account['post_include_media'] = False
+
+
     def set_common_opts(self, account):
         '''
         Set options common to all clients
+        This should only entail setting any defaults or changing any
+        formats that need such or data manipulations required
         :param account:
         '''
 
+        # Failsafe - should already have been initialized...
```

Comment:
Wouldn't be better to raise an `Exception` instead?

Reply:
It would be an exception in the "normal" case, but not when run via a test.  Is it worth it to put the logic in to test for that condition (and not raise an exception in that case)?

Reply:
OK, I didn't get it. I think updating the comment by adding something like "... except during testing" would be enough :)